app/src/main/contrib/components/Responsibilities/Expectations/ExpectationsDescription.tsx

```diff
@@ -1,15 +1,68 @@
 import * as React from "react";
-import {ExpectationMapping} from "../../../../shared/models/Generated";
+import {Country, ExpectationMapping, NumberRange} from "../../../../shared/models/Generated";
+import {Popover, PopoverBody} from "reactstrap";
+
+interface State {
+    showCountries: boolean;
+}
+
+interface Props {
+    countries: Country[]
+    targetKey: string;
+}
+
+export class CountriesList extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+
+        this.toggle = this.toggle.bind(this);
+        this.state = {
+            showCountries: false
+        };
+    }
+
+    toggle() {
+        this.setState({
+            showCountries: !this.state.showCountries
+        });
+    }
+
+    render() {
+        const countriesList = this.props.countries.map(c => c.name).join(", ");
+
+        return <span><a href="#" id={this.props.targetKey}
+                       onClick={this.toggle}>view list</a>
+            <Popover placement="right" isOpen={this.state.showCountries} target={this.props.targetKey}
+                     toggle={this.toggle}>
+                <PopoverBody>{countriesList}</PopoverBody>
+            </Popover></span>
+    }
+}
+
+export class ExpectationsDescription extends React.Component<ExpectationMapping, State> {
 
-export class ExpectationsDescription extends React.PureComponent<ExpectationMapping> {
     render(): JSX.Element {
+
+        const expectation = this.props.expectation;
+        const numCountries = expectation.countries.length;
+        const numYears = expectation.years.maximum_inclusive - expectation.years.minimum_inclusive + 1;
+        const numAges = expectation.ages.maximum_inclusive - expectation.ages.minimum_inclusive + 1;
+        const numCohorts = expectation.cohorts.maximum_birth_year - expectation.cohorts.minimum_birth_year + 1;
```

Comment:
What if one (or both) of the cohorts values is null? I'm also not sure we need to list the number of cohorts - it makes less sense than the others. See my suggested output in another comment below.